src/design-system/components/Box/Box.tsx

```diff
@@ -0,0 +1,177 @@
+import { flatten, mapValues } from 'lodash';
+import React, { forwardRef, ReactNode, useMemo } from 'react';
+import { View } from 'react-native';
+import { BackgroundColor } from '../../color/palettes';
+import { negativeSpace, NegativeSpace, space, Space } from '../../layout/space';
+import { BackgroundProvider } from '../BackgroundProvider/BackgroundProvider';
+import type * as Polymorphic from './polymorphic';
+
+const fraction = (numerator: number, denominator: number) =>
+  `${(numerator * 100) / denominator}%`;
+
+const widths = {
+  '1/2': fraction(1, 2),
+  '1/3': fraction(1, 3),
+  '1/4': fraction(1, 4),
+  '1/5': fraction(1, 5),
+  '2/3': fraction(2, 3),
+  '2/5': fraction(2, 5),
+  '3/4': fraction(3, 4),
+  '3/5': fraction(3, 5),
+  '4/5': fraction(4, 5),
+  full: '100%', // eslint-disable-line prettier/prettier
+} as const;
+
+export interface BoxProps {
+  alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch';
+  background?: BackgroundColor;
+  children?: ReactNode;
+  flexBasis?: 0;
+  flexDirection?: 'row' | 'column';
```

Comment:
Quick search shows we might also need `| 'row-reverse'` (from src/components/token-info/TokenInfoBalanceValue.js)